client/game.js

```diff
@@ -87,43 +113,73 @@
     // Networking
     //
 
-    var handleMessage = function(msg) {
+    var handleWelcome = function(msg) {
         // TODO do stuff~
     };
 
-    var sendMessage = function(msg) {
-        msg = JSON.stringify(msg);
-        socket.send(msg);
+    var handleGoAway = function(msg) {
+        alert('Server terminated our connection: ' + msg.reason);
```

Comment:
We should  impl a better logging system. Perhaps writing to some hidden div that we can have a ~secret~ debug thing to show. This is fine for now, but open an issue for it to be addressed properly at some point